app/version.go

```diff
@@ -109,3 +125,51 @@ func NewBuildInfo() BuildInfo {
 
 	return info
 }
+
+func checkForUpdate(buildInfo *BuildInfo) error {
+	latestInfo, err := fetchLatestVersion()
+	if err != nil {
+		return fmt.Errorf("failed to fetch latest version: %w", err)
```

Comment:
@feihuang after a little more thought I decided to return an error if the client is unable to get the latest version, however the local build info is still printed.